### PR TITLE
Remove ElasticPress dashboard

### PR DESCRIPTION
**Pull request by joshbetz**
We don't need most of the EP dashboard. We'll provide a custom
dashboard for things like health info and we don't want it to
be possible to trigger an index operation from the dashboard.

Note: This only hides the dashboard so far. We'll also want to
explicitly block indexing from the dashboard.

### CODE REVIEW

**Comment by nickdaugherty** [comment]
Should this code go into the main class-elasticsearch.php class? We're currently not separating out ElasticPress specific functionality (though we may in the future).

**Comment by joshbetz** [comment]
@nickdaugherty The longer term vision is to replace the EP dashboard with a simpler VIP dashboard. Some things like index health will go there. It's with that in mind that we started this class. Happy to organize it according to any guidelines you all have been following though.

**Comment by nickdaugherty** [comment]
Ya makes sense - I've found the dashboard pretty useful when setting up ES on sites so far - maybe we can leave it enabled for Automatticians while we work on a replacement?

**Comment by nickdaugherty** [comment]
The EP dashboard hooks into a _lot_ of stuff, it's a shame there isn't a better way to prevent it from running:

https://github.com/10up/ElasticPress/blob/80ae059e58eb9cff2a2fb8de78544c27787ed7b7/includes/dashboard.php#L29-L56

**Comment by elazzabi** [comment]
I still see the dashboard, is there anything else I need to do to test this PR? (other than checking out the code)

Screen Shot 2020-02-13 at 14 47 34

**Comment by elazzabi** [comment]
Tested locally with the help of @joshbetz, works as expected!

File: elasticsearch/includes/classes/class-dashboard.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Automattic\VIP\Elasticsearch;
+
+class Dashboard {
+
+	function __construct() {
+		add_action( 'admin_init', array( __CLASS__, 'admin_init' ) );
```
Comment (nickdaugherty) [inline elasticsearch/includes/classes/class-dashboard.php]: As discussed in the comments, lets be sure that it's left enabled for Automatticians at this point.

Comment (joshbetz) [reply elasticsearch/includes/classes/class-dashboard.php]: Yep, the dashboard still works. We're just hiding the navigation.

Comment (nickdaugherty) [reply elasticsearch/includes/classes/class-dashboard.php]: I mean we should probably just leave the navigation - I've been using it pretty much constantly while setting up sites + testing.

Comment (joshbetz) [reply elasticsearch/includes/classes/class-dashboard.php]: Ok, we can probably just wait to deploy this until we have something to replace it with then.

Comment (nickdaugherty) [reply elasticsearch/includes/classes/class-dashboard.php]: @joshbetz what do you think about 02d31907?

Comment (joshbetz) [reply elasticsearch/includes/classes/class-dashboard.php]: Sweet, that works. I didn't realize it would be so easy 🙃

**Comment by joshbetz** [comment]
r783-vip-go-stacks

**Pull request merged**